js/checkboxgroup/checkboxgroup.test.ts

```diff
@@ -0,0 +1,345 @@
+import { test, describe, assert, afterEach, vi } from "vitest";
+import { cleanup, render } from "@gradio/tootils";
+import event from "@testing-library/user-event";
+import { setupi18n } from "../app/src/i18n";
+
+import CheckboxGroup from "./Index.svelte";
+import type { LoadingStatus } from "@gradio/statustracker";
+
+const loading_status: LoadingStatus = {
+	eta: 0,
+	queue_position: 1,
+	queue_size: 1,
+	status: "complete" as LoadingStatus["status"],
+	scroll_to_output: false,
+	visible: true,
+	fn_index: 0,
+	show_progress: "full"
+};
+
+beforeEach(() => {
+	setupi18n();
+});
+
+afterEach(cleanup);
+
+describe("Values", () => {
+	test("renders correct value when passed as string: single value", async () => {
+		const { getByLabelText } = await render(CheckboxGroup, {
+			value: ["choice_one"],
+			label: "Dropdown",
+			choices: [
+				["Choice One", "choice_one"],
+				["Choice Two", "choice_two"]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+		const item_two = getByLabelText("Choice Two") as HTMLInputElement;
+
+		expect(item_one).toBeChecked();
+		expect(item_two).not.toBeChecked();
+	});
+
+	test("renders correct value when passed as string: multiple values", async () => {
+		const { getByLabelText } = await render(CheckboxGroup, {
+			value: ["choice_one", "choice_three"],
+			label: "Dropdown",
+			choices: [
+				["Choice One", "choice_one"],
+				["Choice Two", "choice_two"],
+				["Choice Three", "choice_three"]
+			]
+		});
+
+		const item_one = getByLabelText("Choice One") as HTMLInputElement;
+		const item_two = getByLabelText("Choice Two") as HTMLInputElement;
+		const item_three = getByLabelText("Choice Three") as HTMLInputElement;
+		assert.equal(item_one.checked, true);
+		assert.equal(item_two.checked, false);
+		assert.equal(item_three.checked, true);
+
+		expect(item_one).toBeChecked();
+		expect(item_two).not.toBeChecked();
+		expect(item_three).toBeChecked();
```

Comment:
Common wisdom says 'one unit test, one assertion' but this is unnecessarily restrictive. Multiple assertions are fine if they are all test a discreet thing. If multiple things are being tested however, they should be broken into new tests.

Overlap in unit tests is fine, they are cheap.